typed-protocols/src/Network/TypedProtocol/Core.hs

```diff
@@ -253,6 +257,15 @@ type family FlipAgency (pk :: PeerKind) where
   FlipAgency AsServer = AsClient
 
 
+-- | Identity function in disguise, only needed because GHC has trouble proving
+-- that @FlipAgency (FlipAgency pk)@ is equal to @pk@.  It turns out to be
```

Comment:
Uh oh. It's possible this could become a very annoying problem. If it does, we can always revert to the way I'd do things: `local` and `remote` type parameters :)